python/cudf/cudf/tests/test_categorical.py

```diff
@@ -677,3 +679,26 @@ def test_add_categories_mixed_error():
 
     with pytest.raises(TypeError):
         gds.cat.add_categories(["a", "bd", "ef"])
+
+
+def test_categorical_equal_compare():
+
+    # Categories are actually different values
+    d1 = CategoricalDtype(categories=[1, 2, 3], ordered=False)
+    d2 = CategoricalDtype(categories=[2, 3, 4], ordered=False)
+
+    assert not d1 == d2
+
+    # Categories are the same, but dtypes different
+    d1 = CategoricalDtype(
+        categories=np.array([1, 2, 3], dtype="int32"), ordered=False
+    )
+    d2 = CategoricalDtype(
+        categories=np.array([1, 2, 3], dtype="int64"), ordered=False
+    )
+
+    assert not d1 == d2
+
```

Comment:
Maybe also have a valid equality case?

Reply:
I was thinking about that. But I'm not sure how to do that without constructing the exact same object twice, and then the situation is trivially equal.

Reply:
We could do something like..
```python
>>> import pandas as pd
>>> x = pd.CategoricalDtype([1,2,3])
>>> y = pd.CategoricalDtype([1, 2, 3])
>>> x == y
True
>>> import cudf
>>> x = cudf.CategoricalDtype([1, 2, 3])
>>> y = cudf.CategoricalDtype([1, 2, 3])
>>> x == y
True
```